File: 2.Product of all elements except one/Solution.js
// Solution class
class Solution {
  static solve(nums) {
    const res = new Array(nums.length).fill(0);
    let runningProduct = 1;
    for (let i = 0; i < nums.length; i++) {
      res[i] = runningProduct;
      runningProduct *= nums[i];
    }
    runningProduct = 1;
    for (let i = nums.length - 1; i >= 0; i--) {
      res[i] *= runningProduct;
      runningProduct *= nums[i];
    }
    return res;
  }
}

// Main
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("", (input) => {
  const nums = input.split(" ").map(Number);
  const result = Solution.solve(nums);
  console.log(result);
  rl.close();
});

File: 2.Product of all elements except one/Solution.py
from typing import List

# Solution Class
class Solution:
    def solve(nums:List[int]) -> List[int]:
        res = [0]*len(nums)
        runningProduct = 1
        for i in range(len(nums)):
            res[i] = runningProduct
            runningProduct *= nums[i]
        runningProduct = 1
        for i in range(len(nums)-1,-1,-1):
            res[i] *= runningProduct
            runningProduct *= nums[i]
        return res

# Main
nums = list(map(int,input().split()))
result = Solution.solve(nums)
print(result)